dd-java-agent/agent-debugger/debugger-el/src/main/java/com/datadog/debugger/el/expressions/ContainsExpression.java

```
@@ -1,15 +1,61 @@
 package com.datadog.debugger.el.expressions;
 
+import com.datadog.debugger.el.EvaluationException;
+import com.datadog.debugger.el.PrettyPrintVisitor;
+import com.datadog.debugger.el.Value;
 import com.datadog.debugger.el.Visitor;
-import com.datadog.debugger.el.values.StringValue;
+import com.datadog.debugger.el.values.CollectionValue;
+import datadog.trace.bootstrap.debugger.el.ValueReferenceResolver;
 
-public class ContainsExpression extends StringPredicateExpression {
-  public ContainsExpression(ValueExpression<?> sourceString, StringValue str) {
-    super(sourceString, str, String::contains, "contains");
+public class ContainsExpression implements BooleanExpression {
+  private final ValueExpression<?> target;
+  private final ValueExpression<?> value;
+
+  public ContainsExpression(ValueExpression<?> target, ValueExpression<?> value) {
+    this.target = target != null ? target : ValueExpression.NULL;
+    this.value = value;
+  }
+
+  @Override
+  public Boolean evaluate(ValueReferenceResolver valueRefResolver) {
+    Value<?> targetValue = target.evaluate(valueRefResolver);
+    if (targetValue.isUndefined()) {
+      throw new EvaluationException(
+          "Cannot evaluate the expression for undefined value", PrettyPrintVisitor.print(this));
+    }
+    if (targetValue.isNull()) {
+      throw new EvaluationException(
+          "Cannot evaluate the expression for null value", PrettyPrintVisitor.print(this));
+    }
+    Value<?> val = value.evaluate(valueRefResolver);
+    if (val.isUndefined() || val.isNull()) {
```

do we want to fail for undefined or null?

For example. maybe the user would like to do contains(array, null). 

I'm not sure about undefined...

---

added support for this